src/frontends/onnx/frontend/src/op/com.microsoft/matmulnbits.cpp

```diff
@@ -45,7 +45,7 @@ ov::OutputVector matmulnbits(const ov::frontend::onnx::Node& node) {
         4);  // required, in docs: number of bits used for weight quantization (default 4)
 
     const uint64_t n_blocks_per_col = (K + block_size - 1) / block_size;
-    const auto blob_size = static_cast<int64_t>(ceil(block_size * bits / 8));
+    const auto blob_size = static_cast<int64_t>(ceil(static_cast<float>(block_size) * bits / 8));
```

Comment:
I don't quite understand why the previous version is not correct. All operands are of int64_t type. Please clarify what values in the case you are fixing.

Reply:
Can we do this as follows:

`
const auto blob_size = (block_size * bits + 7) / 8));
`

Reply:
Yep, will work same way, agree